File: 3705.java
//* @author
import java.util.Scanner;

public class Main {
Scanner cin = new Scanner(System.in);
int n;

public void inPut() {
   n = cin.nextInt();

   reverse();
}

private void reverse() {
   if (n == 1) {
    System.out.println(0);
   } else {
    if (n == 2) {
     System.out.println(1);
     System.out.println(2 + " " + 1 + " " + 0);
    } else {
     if (n % 2 == 0) {
      int m = (n + 1) / 2 + 1;
      System.out.println(m);
      print(n, m);
     } else {
      if(n % 2 != 0) {
       int m = (n + 1) / 2;
       System.out.println(m);
       print(n, m);
      }
     }
    }
   }
}

void print(int n, int m) {
   if (n % 2 != 0) {
    for (int i = 0; i < m - 1; i++) {
     System.out.println((n / 2 + i + 1) + " " + 2 + " " + i);
    }
    System.out.println(1 + " " + (n - 1) / 2 + " " + (n - 1) / 2);

   } else {
    for (int i = 0; i < m - 2; i++) {
     System.out.println((n / 2 + i + 1) + " " + 2 + " " + (i + 1));
    }
    System.out.println(2 + " " + (n - 1) / 2 + " " + (n) / 2);
    System.out.println(1 + " " + 1 + " " + (n - 1));

   }
}

public static void main(String[] args) {
   new Main().inPut();
}

}

File: 3705.cc
#include <iostream>
using namespace std;
void solve(int n) {
  int x = (n + 1) / 2 - 1;
  int y = n;
  for (int i = 0; i < x; ++i) {
    cout << n / 2 << " " << 2 << " " << y - 2 - i << endl;
    n -= 2;
  }
  cout << "2 " << x << " " << x + 1 << endl;
}

int main() {
  int n;
  cin >> n;
  if (n == 1) {
    cout << 0 << endl;
    return 0;
  }
  if (n == 2) {
    cout << "1" << endl;
    cout << "1 1 1" << endl;
    return 0;
  }
  if (n % 2 != 0) {
    cout << (n + 1) / 2 << endl;
    solve(n);
  } else {
    cout << n / 2 + 1 << endl;
    solve(n - 1);
    cout << 1 << " " << n - 1 << " 1" << endl;
  }

  return 0;
}
